File: src/main/java/com/socialhobbies/shuserservice/entity/Role.java
package com.socialhobbies.shuserservice.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import org.hibernate.annotations.NaturalId;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "role")
public class Role  {
	@Id
	@Column (name = "role_id")
	@GeneratedValue (strategy =  GenerationType.IDENTITY)
	private Long id;

	@Column(name = "name")
	@Enumerated (EnumType.STRING)
	@NaturalId
	private ERoleName name;

	@ManyToMany(mappedBy = "roles", fetch = FetchType.LAZY)
	@JsonIgnore
	private Set<User> users = new HashSet<>();

	public Role() {}
	
	public Role(ERoleName role) {
		this.name = role;
	}

	public boolean isAdminRole() {
		return this.name.equals(ERoleName.ROLE_ADMIN);
  }

	public Long getId() {
		return this.id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public ERoleName getRole() {
		return this.name;
	}

	public void setRole(ERoleName role) {
		this.name = role;
	}

	public Set<User> getUsers() {
		return this.users;
	}

	public void setUsers(Set<User> users) {
		this.users = users;
	}

	@Override
	public String toString() {
		return "{" +
			" id='" + getId() + "'" +
			", role='" + getRole() + "'" +
			", users='" + getUsers() + "'" +
			"}";
	}

} 

File: src/main/java/com/smarthome/shuserservice/entity/Role.kt
package com.smarthome.shuserservice.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import org.hibernate.annotations.NaturalId
import java.util.*
import javax.persistence.*

@Entity
class Role(
    @Id @Column(name = "ROLE_ID")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(name = "ROLE_NAME")
    @Enumerated(EnumType.STRING)
    @NaturalId
    var role: ERoleName? = null,

    @ManyToMany(mappedBy = "roles", fetch = FetchType.LAZY)
    @JsonIgnore
    var userOlds: MutableSet<UserOld> = HashSet()

) {
    fun isAdminRole(): Boolean = this.role == ERoleName.ROLE_ADMIN

}
